onnxscript/converter.py

```diff
@@ -406,6 +401,20 @@ def translate_attr(self, attr_name, expr):
             val = self.lookup(expr.id, debuginfo.DebugInfo(expr, self))
             if isinstance(val, values.AttrRef):
                 return self.ir_builder.attr_ref(attr_name, val.value, val.typeinfo)
+            if isinstance(val, irbuilder.Function):
+                # Check that outer-scope variables referenced by function have same value
+                # at function-definition site and use-as-attribute site, to avoid errors.
+                for pyvar, previous in val.outer_scope_variables:
+                    current = self.lookup(pyvar, debuginfo.DebugInfo(expr, self))
+                    if current.value != previous.value:
```

Comment:
Is there a unit test to check this exception?

Reply:
Yes, see `test_outer_scope_redefinition`.